bson/_cbsonmodule.c

```diff
@@ -2948,7 +2948,7 @@ static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* a
             goto fail;
          }
 
-        if (view.len < size) {
+        if (view.len < (int32_t)size) {
```

Comment:
Shouldn't this check and `if (string[size - 1]) {` below be done  up above directly after parsing the `size`?

Reply:
Yeah that makes sense to me, fixed.